crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -207,7 +206,12 @@ fn index_stack_trace_to_function_name_stack_trace(
             // Remove suffix in case of loop function e.g. `[expr36]`.
             let func_name = re_loop_func.replace(sierra_func_name, "");
             // Remove parameters from monomorphised Cairo generics e.g. `<felt252>`.
-            re_monomorphization.replace(&func_name, "").to_string()
+            if split_generics {
+                func_name
+            } else {
+                re_monomorphization.replace(&func_name, "")
+            }
+            .to_string()
```

Comment:
This shouldn't be responsibility of idx mapping imo. We should always split generics and perform merging separately.

Reply:
I will move the regex related logic to another function performing `UserFunctionIdx` -> `String` conversion and move the logic inside it - I will need this function for https://github.com/foundry-rs/starknet-foundry/issues/2118 anyways (in the mappings from compiler function are always in form of absolute paths, I need a function to convert it to it)

Reply:
True, refactored a whole bunch of other things together with it